tests/test_haarpsi.py

```diff
@@ -0,0 +1,60 @@
+import torch
+from piq import haarpsi, HaarPSILoss
+from PIL import Image
+import numpy as np
+from typing import Tuple
+
+
+# ================== Test function: `haarpsi` ==================
+def test_haarpsi_to_be_one_for_identical_inputs(input_tensors: Tuple[torch.Tensor, torch.Tensor], device: str) -> None:
+    prediction, _ = input_tensors
+    index = haarpsi(prediction.to(device), prediction.to(device), data_range=1., reduction='none')
+    index_255 = haarpsi(prediction.to(device) * 255, prediction.to(device) * 255, data_range=255, reduction='none')
+    assert torch.isclose(index, torch.ones_like(index, device=device)).all(), \
+        f'Expected index to be equal 1, got {index}'
+    assert torch.isclose(index_255, torch.ones_like(index_255, device=device)).all(), \
+        f'Expected index to be equal 1, got {index_255}'
+
+
+def test_haarpsi_symmetry(input_tensors: Tuple[torch.Tensor, torch.Tensor], device: str) -> None:
+    prediction, target = input_tensors
+    result = haarpsi(prediction.to(device), target.to(device), data_range=1., reduction='none')
+    result_sym = haarpsi(target.to(device), prediction.to(device), data_range=1., reduction='none')
+    assert torch.isclose(result_sym, result).all(), f'Expected the same results, got {result} and {result_sym}'
+
+
+def test_haarpsi_zeros_ones_inputs(device: str) -> None:
+    zeros = torch.zeros(1, 3, 256, 256, device=device)
+    ones = torch.ones(1, 3, 256, 256, device=device)
+    haarpsi_ones = haarpsi(ones, ones, data_range=1.)
+    assert torch.isfinite(haarpsi_ones).all(), f'Expected finite value for ones tensos, got {haarpsi_ones}'
+    haarpsi_zeros_ones = haarpsi(zeros, ones, data_range=1.)
+    assert torch.isfinite(haarpsi_zeros_ones).all(), \
+        f'Expected finite value for zeros and ones tensos, got {haarpsi_zeros_ones}'
+
+
+def test_haarpsi_compare_with_matlab(device: str) -> None:
+    prediction = torch.tensor(np.array(Image.open('tests/assets/I01.BMP'))).permute(2, 0, 1)
+    target = torch.tensor(np.array(Image.open('tests/assets/i01_01_5.bmp'))).permute(2, 0, 1)
+    predicted_score = haarpsi(prediction.to(device), target.to(device), data_range=255, reduction='none')
+    target_score = torch.tensor([0.7170]).to(predicted_score)
```

Comment:
Could you provide the target score with higher precision (even if it does not reduce `atol`)? Another option can be to compare with the original python version.

Reply:
Done 👍